src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const arrayFromDate = date.split(oldSeparator);
+  const result = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    for (let j = 0; j < fromFormat.length; j++) {
+      if (toFormat[i] === 'YYYY' && fromFormat[j] === 'YY') {
+        fromFormat[j] = 'YYYY';
+
+        if (arrayFromDate[j] < 30) {
+          arrayFromDate[j] = '20' + arrayFromDate[j];
+        } else {
+          arrayFromDate[j] = '19' + arrayFromDate[j];
+        }
+      }
```

Comment:
The condition `toFormat[i] === 'YYYY' && fromFormat[j] === 'YY'` is intended to convert a two-digit year to a four-digit year. However, modifying `fromFormat[j]` directly may lead to incorrect behavior in subsequent iterations. Consider using a separate variable to handle format conversions without altering the original `fromFormat` array.